internal/pkg/pluginmanager/manager.go

```diff
@@ -33,8 +34,39 @@ func DownloadPlugins(conf *configloader.Config) error {
 			}
 			continue
 		}
-		log.Printf("Plugin: %s already exists, no need to download.", pluginFileName)
+		// check md5
+		dup, err := checkFileMD5(filepath.Join(pluginDir, pluginFileName), dc, pluginFileName, tool.Version)
+		if err != nil {
+			return err
+		}
+
+		if dup {
+			log.Printf("Plugin: %s already exists, no need to download.", pluginFileName)
+			continue
+		}
+
+		log.Printf("Plugin: %s changed and will be overrided.", pluginFileName)
+		os.Remove(filepath.Join(pluginDir, pluginFileName))
+		if err = dc.download(pluginDir, pluginFileName, tool.Version); err != nil {
+			return err
+		}
 	}
 
 	return nil
 }
+
+func checkFileMD5(file string, dc *PbDownloadClient, pluginFileName string, version string) (bool, error) {
+	localmd5, err := LocalContentMD5(file)
+	if err != nil {
+		return false, err
+	}
+	remotemd5, err := dc.fetchContentMD5(pluginFileName, version)
+	if err != nil {
+		return false, err
+	}
+
+	if strings.Compare(localmd5, remotemd5) == 0 {
```

Comment:
It is usually clearer and always faster to use the built-in
// string comparison operators ==, <, >, and so on.